src/lib.rs

```diff
@@ -175,6 +215,56 @@ impl Client {
         self.fetch::<Hotspot>(&format!("/hotspots/{}", address))
     }
 
+    /// Get details for a given hotspot address
+    pub fn get_hotspot_rewards(
+        &self,
+        address: &str,
+        min_time: Option<&str>,
+        max_time: Option<&str>,
+        cursor: Option<&str>,
+    ) -> error::Result<(Vec<Reward>, Option<String>)> {
+        let base = Url::parse(&format!("{}/", self.base_url)).unwrap();
+        let mut request_url = base.join(&format!("hotspots/{}/rewards", address)).unwrap();
+        for (k, v) in [
```

Comment:
Try using `Url::parse_with_params` after creating a vec of options that have values.

Reply:
Nice, I've added a `build_url` private helper to encapsulate that and the "optional param" logic, swapping out the previous implementation.